sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -84,8 +99,24 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   private async getToken(options: GetTokenOptions): Promise<string | undefined> {
     let accessToken = this.tokenCache.getCachedToken();
     if (accessToken === undefined) {
-      accessToken = (await this.credential.getToken(this.scopes, options)) || undefined;
-      this.tokenCache.setCachedToken(accessToken);
+      // Waiting for the next refresh (or forcefully triggering a new refresh)
+      // only if the cache is unable to retrieve the access token,
+      // which means that it has expired, or it has never been set.
+      const refreshPromise = this.tokenRefresher.forcedRefresh(options);
+      if (refreshPromise !== null) {
```

Comment:
yeah having to do this check is ugly

Reply:
Here's a much better approach with some small tweaks! How come I didn't think of this earlier? 😓 well, it's here now! 😁 https://github.com/Azure/azure-sdk-for-js/pull/10085/commits/16c88f5e2431c0f1a64ec1b44656d51ec1cbb709